Trouble building HelloCollision

See [discussion](https://hub.jmonkeyengine.org/t/trouble-running-hellocollision-with-maven-artifacts-form-sbt/38875/20) for the issue on the JME forums. The [Java example works with Gradle](https://github.com/bbarker/HelloCollision-on-Gradle):

```
brandon@brandon-750-170se:/media/brandon/portadrive/projects/HelloCollision-on-Gradle$ gradle build
:compileJava
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-core/3.1.0-stable/jme3-core-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-bullet/3.1.0-stable/jme3-bullet-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-bullet-native/3.1.0-stable/jme3-bullet-native-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-desktop/3.1.0-stable/jme3-desktop-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-jogg/3.1.0-stable/jme3-jogg-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-lwjgl/3.1.0-stable/jme3-lwjgl-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-plugins/3.1.0-stable/jme3-plugins-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-terrain/3.1.0-stable/jme3-terrain-3.1.0-stable.pom
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-core/3.1.0-stable/jme3-core-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-bullet/3.1.0-stable/jme3-bullet-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-bullet-native/3.1.0-stable/jme3-bullet-native-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-desktop/3.1.0-stable/jme3-desktop-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-jogg/3.1.0-stable/jme3-jogg-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-lwjgl/3.1.0-stable/jme3-lwjgl-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-plugins/3.1.0-stable/jme3-plugins-3.1.0-stable.jar
Download https://jcenter.bintray.com/org/jmonkeyengine/jme3-terrain/3.1.0-stable/jme3-terrain-3.1.0-stable.jar
:processResources UP-TO-DATE
:classes
:jar
:assemble
:compileTestJava UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:test UP-TO-DATE
:check UP-TO-DATE
:build

BUILD SUCCESSFUL

Total time: 14.239 secs
```

And it runs using:

```
gradle -PmainClass=jme3test.helloworld.HelloCollision execute
```

After doing a clean, trying out other versions of the Scala compiler, and switching back (and maybe other things along the way), it is now broken again with a different `NoClassDefFoundError`:

```
[error] Oct 08, 2017 6:51:18 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
[error] WARNING: Pausing audio device not supported.
[error] Oct 08, 2017 6:51:18 PM com.jme3.app.LegacyApplication handleError
[error] SEVERE: Uncaught exception thrown in Thread[jME3 Main,5,main]
[error] java.lang.NoClassDefFoundError: com/bulletphysics/collision/broadphase/OverlapFilterCallback
[error] 	at com.jme3.bullet.BulletAppState.startPhysics(BulletAppState.java:164)
[error] 	at com.jme3.bullet.BulletAppState.stateAttached(BulletAppState.java:211)
[error] 	at com.jme3.app.state.AppStateManager.attach(AppStateManager.java:133)
[error] 	at tutorial.HelloCollision.simpleInitApp(HelloCollision.scala:65)
[error] 	at com.jme3.app.SimpleApplication.initialize(SimpleApplication.java:220)
[error] 	at com.jme3.system.lwjgl.LwjglAbstractDisplay.initInThread(LwjglAbstractDisplay.java:130)
[error] 	at com.jme3.system.lwjgl.LwjglAbstractDisplay.run(LwjglAbstractDisplay.java:211)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] Caused by: java.lang.ClassNotFoundException: com.bulletphysics.collision.broadphase.OverlapFilterCallback
[error] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[error] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[error] 	... 8 more
[error] 

```

The bad news is, this time, running the gradle project does not seem to act as a workaround.

--- a/.gitignore
+++ b/.gitignore
@@ -9,3 +9,5 @@ data
 .idea
 *~
 .dotty-ide*
+
+*.dll

--- a/build.sbt
+++ b/build.sbt
@@ -49,12 +49,12 @@ lazy val root = (project in file("."))
 
 
 lazy val jmonkeyDeps = Seq(
-      "org.jmonkeyengine" % "jme3-core" % "3.1.0-stable" exclude("org.jmonkeyengine", "jme3-testdata")
-      ,"org.jmonkeyengine" % "jme3-examples" % "3.1.0-stable" exclude("org.jmonkeyengine", "jme3-testdata")
-      ,"org.jmonkeyengine" % "jme3-bullet" % "3.1.0-stable" exclude("org.jmonkeyengine", "jme3-testdata")
-      ,"org.jmonkeyengine" % "jme3-bullet-native" % "3.1.0-stable" exclude("org.jmonkeyengine", "jme3-testdata")
-//      ,"org.jmonkeyengine" % "jme3-jbullet" % "3.1.0-stable" exclude("org.jmonkeyengine", "jme3-testdata")
-//    ,"org.jmonkeyengine" % "jme3-testdata" % "3.1.0-stable" from "file://lib/jme3-testdata-3.1.0.jar"
-      ,"org.lwjgl.lwjgl" % "lwjgl_util" % "2.9.3"
-//    ,"org.specs2" %% "specs2-core" % "3.8.9" % "it,test"
+    "org.jmonkeyengine" % "jme3-bullet" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.jmonkeyengine" % "jme3-bullet-native" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.jmonkeyengine" % "jme3-core" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.jmonkeyengine" % "jme3-desktop" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.jmonkeyengine" % "jme3-examples" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.jmonkeyengine" % "jme3-lwjgl" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.jmonkeyengine" % "jme3-terrain" % "3.2.1-stable" exclude("org.jmonkeyengine", "jme3-testdata") exclude("stack-alloc", "stack-alloc") exclude("jbullet", "jbullet")
+    ,"org.lwjgl.lwjgl" % "lwjgl_util" % "2.9.3"
 )

--- a/src/main/scala/tutorial/HelloCollision.scala
+++ b/src/main/scala/tutorial/HelloCollision.scala
@@ -124,7 +124,7 @@ class HelloCollision extends SimpleApplication with ActionListener {
       case Right => characterDirection = characterDirection.copy(right = value)
       case Up => characterDirection = characterDirection.copy(up = value)
       case Down => characterDirection = characterDirection.copy(down = value)
-      case Jump =>  if (value) { player.jump(new Vector3f(0,20f,0));}
+      case Jump =>  if (value) { player.jump(new Vector3f(0,20f,0)) }
   }
  
   /**

--- a/src/main/scala/tutorial/HelloCollision.scala
+++ b/src/main/scala/tutorial/HelloCollision.scala
@@ -79,7 +79,7 @@ class HelloCollision extends SimpleApplication with ActionListener {
 
     player.setJumpSpeed(20)
     player.setFallSpeed(30)
-    player.setGravity(30)
+    player.setGravity(new Vector3f(0,-30f,0))
     player.setPhysicsLocation(new Vector3f(0, 10, 0))
  
     // We attach the scene and the player to the rootNode and the physics space,
@@ -124,7 +124,7 @@ class HelloCollision extends SimpleApplication with ActionListener {
       case Right => characterDirection = characterDirection.copy(right = value)
       case Up => characterDirection = characterDirection.copy(up = value)
       case Down => characterDirection = characterDirection.copy(down = value)
-      case Jump => player.jump()
+      case Jump =>  if (value) { player.jump(new Vector3f(0,20f,0));}
   }
  
   /**

